src/components/Block.jsx

```diff
@@ -20,20 +21,31 @@ module.exports = React.createClass({
     return app.refine('blockTypes').find(i => i.id === block.type)
   },
 
+  setMenuItems(component) {
+    if ('getMenuItems' in component) {
+      this.setState({ extraMenuItems: component.getMenuItems() })
+    }
+  },
+
+  setComponentRef(component) {
+    this.setMenuItems(component)
+    return 'block'
+  },
```

Comment:
It _is_ confusing. Refs can be defined as functions, and it will store the returned value as the key in the `this.refs` object. I'm taking advantage of the fact that it passes the component here to assign a state property that gets the specific menu items for a block type